### PR TITLE
Warn for pattern matching on tuple literals

### PR DESCRIPTION
**Pull request by giacomocavalieri**

### CODE REVIEW

File: compiler-core/src/warning.rs [hunk 1]
```diff
@@ -598,6 +598,25 @@ the same values.\n"
                         }),
                     }
                 }
+
+                type_::Warning::CaseMatchOnLiteralTuple { location } => Diagnostic {
+                    title: "Case expression with literal tuple as subject".into(),
```
Comment (lpil) [inline compiler-core/src/warning.rs:603]: How about `Redundant tuple`?

**Review by lpil — COMMENTED** [review]
Nice, thank you! Looks like the build is faiing

File: compiler-core/src/warning.rs (hunk 1 above)
Comment (giacomocavalieri) [reply compiler-core/src/warning.rs:603]: Way better! Thanks for the suggestion 💖

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,8 @@
   returned by blocks.
 - Fix a bug where Erlang code generation would not properly handle list case patterns
   with no head and a spread tail.
+- The compiler will now raise a warning ig you're pattern matching on tuple
```
Comment (lpil) [inline CHANGELOG.md:26]:
```suggestion
- The compiler will now raise a warning if you're pattern matching on tuple
```

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**